deeplearning4j-core/src/main/java/org/deeplearning4j/eval/Evaluation.java

```diff
@@ -184,55 +184,55 @@ public void eval(INDArray realOutcomes, INDArray guesses) {
 
         // For each row get the most probable label (column) from prediction and assign as guessMax
         // For each row get the column of the true label and assign as currMax
-        int nCols = realOutcomes.columns();
-        for (int i = 0; i < realOutcomes.rows(); i++) {
-            INDArray currRow = realOutcomes.getRow(i);
-            INDArray guessRow = guesses.getRow(i);
-
-
-            int currMax;
-            int guessMax;
-            if( nCols == 1){
-                //Binary (single variable) case
-                if(currRow.getDouble(i) == 0.0) currMax = 0;
-                else currMax = 1;
 
-                if(guessRow.getDouble(i) <= 0.5 ) guessMax = 0;
-                else guessMax = 1;
+        int nCols = realOutcomes.columns();
+        int nRows = realOutcomes.rows();
+
+        if (nCols == 1) {
+            INDArray binaryGuesses = guesses.gt(0.5);
+            int tp = binaryGuesses.mul(realOutcomes).sumNumber().intValue();
+            int fp = binaryGuesses.mul(-1.0).addi(1.0).muli(realOutcomes).sumNumber().intValue();
+            int fn = binaryGuesses.mul(realOutcomes.mul(-1.0).addi(1.0)).sumNumber().intValue();
+            int tn = nRows - tp - fp - fn;
+
+            confusion.add(1, 1, tp);
+            confusion.add(1, 0, fn);
+            confusion.add(0, 1, fp);
+            confusion.add(0, 0, tn);
+
+            truePositives.incrementCount(1, tp);
+            falsePositives.incrementCount(1, fp);
+            falseNegatives.incrementCount(1, fp);
+            trueNegatives.incrementCount(1, tp);
+
+            truePositives.incrementCount(0, tn);
+            falsePositives.incrementCount(0, fn);
+            falseNegatives.incrementCount(0, fn);
+            trueNegatives.incrementCount(0, tn);
```

Comment:
@agibsonccc I'd appreciate if you could take closer look at these 8 lines above. It's passing the test (which doesn't cover everything), but I'm not sure if it's doing the right thing.